src/Pos/Update/Poll/Pure.hs

```diff
@@ -0,0 +1,117 @@
+-- | Pure Poll
+
+module Pos.Update.Poll.Pure
+       ( PurePoll (..)
+       , evalPurePollWithLogger
+       , execPurePollWithLogger
+       , runPurePollWithLogger
+       ) where
+
+import           Universum
+
+import           Control.Lens               (at, to, (%=), (.=))
+import qualified Data.HashMap.Strict        as HM
+import           System.Wlog                (CanLog, HasLoggerName (..), LogEvent,
+                                             NamedPureLogger, logWarning, runNamedPureLog)
+
+import           Pos.Crypto                 (hash)
+import           Pos.Types                  (SoftwareVersion (..))
+import           Pos.Update.Poll.Class      (MonadPoll (..), MonadPollRead (..))
+import qualified Pos.Update.Poll.PollState  as Poll
+import           Pos.Update.Poll.Types      (BlockVersionState (..),
+                                             DecidedProposalState (..),
+                                             UndecidedProposalState (..),
+                                             cpsSoftwareVersion, propStateToEither)
+
+newtype PurePoll a = PurePoll
+    { getPurePoll :: NamedPureLogger (State Poll.PollState) a
+    } deriving (Functor, Applicative, Monad, CanLog, HasLoggerName)
+
+instance HasLoggerName Identity where
```

Comment:
It's needed for this line: https://github.com/input-output-hk/cardano-sl/pull/240/files/df98119710255b42bacab1ec9448d7194458c255#diff-b8b5a7ecdf02e93f085a2de817eba3f4R37

`runNamedPureLog` requires `HasLoggerNameIdentity`. I'm going to add this to `log-warper`, but until it gets merged to `master` this'll do. It's also in the module where `PureToss` is defined.